src/components/RestroPage/RestroDetails/RestroDetails.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from "react";
+import "./RestroDetails.css"
+import list from "./constants/details.list"
+import Menu from "./Menu"
+ class RestroDetails extends Component {
+  
+    constructor(){
+        super();
+        this.state=
+        {
+            cart: [],
+            dishList: list
+        }
+    }
+  updateDishInCart(detailCopy){
+   
+    return (current)=>{
+      if(current.name === detailCopy.name)
+      {
+        current.quantity=current.quantity+1;
+      }
+    }
+     
+    
+  }
+  
+  isPresentInCart(detail,isInCart){
+      return (dish)=>{
+        if(detail.name===dish.name){
+            isInCart.isTrue=1;
+        }
+      }
+  }
+  handleClickLogic(detail,newCart){
+   
+    let detailCopy={...detail};
+    let isInCart={
+      isTrue:0
+    };
+    newCart.map(this.isPresentInCart(detail,isInCart));
+    console.log(isInCart);
+    if(isInCart.isTrue===0)
+    {
+     detailCopy.quantity=1;
+     newCart.push(detailCopy);
+    }
+    else
+   {
+      newCart.map(this.updateDishInCart(detailCopy));
+   }
+    
+  }
+
+
+  handleClick(detail)
+  {
+    let newCart=[...(this.state.cart)];
+
+    this.handleClickLogic(detail,newCart);
```

Comment:
First thing
1. Try to write all the logic inside a helper file and not the component. 
2. A function should not mutate its arguments
3. const newCartValue = getNewCartValue(cart)